### PR TITLE
Base58 and multibase support

### PR DESCRIPTION
**Pull request by DenisRybas**
Base58 and multibase support for secrets and verificaton methods added

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -135,6 +153,35 @@ impl From<serde_json::Error> for Error {
     }
 }
 
+impl From<bs58::decode::Error> for Error {
+    fn from(err: bs58::decode::Error) -> Self {
+        match err {
+            bs58::decode::Error::BufferTooSmall => {
+                Error::msg(ErrorKind::InvalidState, err.to_string())
+            }
+            bs58::decode::Error::InvalidCharacter {
```
Comment (ashcherbakov) [inline src/error.rs:162]: I believe `InvalidCharacter` and `NonAsciiCharacter` are quite expected errors during decoding meaning that the message is Malformed. So, I think it should generate Malformed error kind, not InvalidState.

File: src/error.rs [hunk 2]
```diff
@@ -135,6 +153,35 @@ impl From<serde_json::Error> for Error {
...
+    }
+}
+
+impl From<bs58::encode::Error> for Error {
+    fn from(err: bs58::encode::Error) -> Self {
+        match err {
+            bs58::encode::Error::BufferTooSmall => {
+                Error::msg(ErrorKind::InvalidState, err.to_string())
+            }
+            _ => Error::msg(ErrorKind::Malformed, err.to_string()),
```
Comment (ashcherbakov) [inline src/error.rs:180]: I believe we should throw Malformed only when decoding / de-serializing, that is when we have some output that we don't control. 
When encoding / serializing, we don't expect any "malformed" input, as we created that input in our code. So, if something unexpected happens during encoding / serialization, that is probably always `InvalidState`, not `Malformed`.

File: src/utils/did.rs [hunk 3]
```diff
@@ -115,38 +294,357 @@ impl AsKnownKeyPair for Secret {
...
+            0x1300 => Ok(Codec::Ed25519Priv),
+            _ => Err(err_msg(ErrorKind::IllegalArgument, "Unsupported prefix")),
+        };
     }
 }
 
+fn _from_multicodec(value: &Vec<u8>) -> Result<(Codec, &[u8])> {
+    let mut val: Cursor<Vec<u8>> = Cursor::new(value.clone());
+    let prefix_int = val.read_unsigned_varint_32().kind(
+        ErrorKind::IllegalArgument,
```
Comment (ashcherbakov) [inline src/utils/did.rs:456]: I believe varint's `read_unsigned_varint_32` and `write_unsigned_varint_32` throw IO Error which means that we can not read/write, and not that the prefix is invalid. So we should use `InvalidState`error kind, not `IllegalArgument`.

Comment (DenisRybas) [reply src/utils/did.rs]: fixed

File: src/error.rs (hunk 2 above)
Comment (DenisRybas) [reply src/error.rs]: fixed

File: src/error.rs (hunk 1 above)
Comment (DenisRybas) [reply src/error.rs]: fixed

**Review by ashcherbakov — APPROVED** [review]

**Review by spivachuk — APPROVED** [review]

**Pull request merged**